internal/service/encryptionatrestprivateendpoint/resource_test.go

```diff
@@ -1,29 +1,146 @@
-// 		//		CheckDestroy:             checkDestroyEncryptionatrestprivateendpoint,
-// 		Steps: []resource.TestStep{ // TODO: verify updates and import in case of resources
-// 			//			{
-// 			//				Config: encryptionatrestprivateendpointConfig(),
-// 			//				Check:  encryptionatrestprivateendpointAttributeChecks(),
-// 			//			},
-// 			//          {
-// 			//				Config: encryptionatrestprivateendpointConfig(),
-// 			//				Check:  encryptionatrestprivateendpointAttributeChecks(),
-// 			//			},
-// 			//			{
-// 			//				Config:            encryptionatrestprivateendpointConfig(),
-// 			//				ResourceName:      resourceName,
-// 			//				ImportStateIdFunc: checkEncryptionatrestprivateendpointImportStateIDFunc,
-// 			//				ImportState:       true,
-// 			//				ImportStateVerify: true,
-// 		},
-// 	},
-// 	)
-// }
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+	"go.mongodb.org/atlas-sdk/v20240805001/admin"
+)
+
+func TestAccEncryptionAtRestPrivateEndpoint_basic(t *testing.T) {
+	resource.ParallelTest(t, *basicTestCase(t))
+}
+
+func basicTestCase(tb testing.TB) *resource.TestCase {
+	tb.Helper()
+	acc.SkipTestForCI(tb) // needs Azure configuration
+	var (
+		resourceName  = "mongodbatlas_encryption_at_rest_private_endpoint.test"
+		projectID     = os.Getenv("MONGODB_ATLAS_PROJECT_ID")
+		azureKeyVault = &admin.AzureKeyVault{
+			Enabled:                  conversion.Pointer(true),
+			RequirePrivateNetworking: conversion.Pointer(true),
+			AzureEnvironment:         conversion.StringPtr("AZURE"),
+			ClientID:                 conversion.StringPtr(os.Getenv("AZURE_CLIENT_ID")),
+			SubscriptionID:           conversion.StringPtr(os.Getenv("AZURE_SUBSCRIPTION_ID")),
+			ResourceGroupName:        conversion.StringPtr(os.Getenv("AZURE_RESOURCE_GROUP_NAME")),
+			KeyVaultName:             conversion.StringPtr(os.Getenv("AZURE_KEY_VAULT_NAME")),
+			KeyIdentifier:            conversion.StringPtr(os.Getenv("AZURE_KEY_IDENTIFIER")),
+			Secret:                   conversion.StringPtr(os.Getenv("AZURE_SECRET")),
+			TenantID:                 conversion.StringPtr(os.Getenv("AZURE_TENANT_ID")),
+		}
+		region = os.Getenv("AZURE_PRIVATE_ENDPOINT_REGION")
+	)
+
+	return &resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(tb) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: configPrivateEndpointAzureBasic(projectID, azureKeyVault, region),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttr(resourceName, "status", "PENDING_ACCEPTANCE"),
```

Comment:
import from `retry_state.RetryStrategyPendingAcceptanceState` or hard-code `PENDING_ACCEPTANCE` ?

Reply:
adjusting to use the variable